src/exporters/vhi.py

```diff
@@ -0,0 +1,220 @@
+        ftp.login()
+        ftp.cwd('/pub/corp/scsb/wguo/data/Blended_VH_4km/VH/')
+
+        # download each filename using this FTP object
+        for raw_filename in legit_filenames:
+            output_filename = (
+                make_filename(raw_folder, raw_filename, dataset='vhi')
+            )
+            download_file_from_ftp(ftp, raw_filename, output_filename)
+
+    return
+
+
+class VHIExporter(BaseExporter):
+    """Exports Vegetation Health Index from NASA site
+
+    ftp.star.nesdis.noaa.gov
+    """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+        super().__init__(data_folder)
+
+    @staticmethod
+    def get_ftp_filenames() -> List:
+        """  get the filenames containing VHI """
+        with FTP('ftp.star.nesdis.noaa.gov') as ftp:
+            ftp.login()
+            ftp.cwd('/pub/corp/scsb/wguo/data/Blended_VH_4km/VH/')
+
+            # append the filenames to a list
+            listing: List = []
+            ftp.retrlines("LIST", listing.append)
+            # extract the filename
+            filepaths = [f.split(' ')[-1] for f in listing]
+            # extract only the filenames of interest
+            vhi_files = [f for f in filepaths if ".VH.nc" in f]
+
+        return vhi_files
+
+    # @staticmethod
+    def chunks(self, l: List, n: int) -> Generator:
+        # return a generator object which chunks list into sublists of size n
+        # https://chrisalbon.com/python/data_wrangling/break_list_into_chunks_of_equal_size/
+        # For item i in a range that is a length of l,
+        for i in range(0, len(l), n):
+            # Create an index range for l of n items:
+            yield l[i:i + n]
+
+    def save_errors(self, outputs: List) -> None:
+        print("\nError: ", [errors for errors in outputs if errors is not None])
+
+        # save the filenames that failed to a pickle object
+        with open(self.raw_folder / 'vhi_export_errors.pkl', 'wb') as f:
+            pickle.dump([error[-1] for error in outputs if error is not None], f)
+
+        return
+
+    def run_parallel(self,
+                     vhi_files: List,
+                     years: List[int] = [yr for yr in np.arange(1981, 2020)]
+                     ) -> List:
+        pool = Pool(processes=100)
+
+        # split the filenames into batches of 100
+        batches = [batch for batch in self.chunks(vhi_files, 100)]
+
+        # run in parallel for multiple file downloads
+        # passing mutliple args using partial https://stackoverflow.com/a/5442981/9940782
+        args = dict(raw_folder=self.raw_folder, years=years)
+        outputs = pool.map(partial(batch_ftp_request, args), batches)
+
+        # write the output (TODO: turn into logging behaviour)
+        print("\n\n*************************")
+        print("VHI Data Downloaded")
+        print("*************************")
+        print("Errors:")
+        pprint([error for error in outputs if error is not None])
```

Comment:
Since the errors are also printed in `save_errors`, they would get printed twice. Can we eliminate one of the print statements?

Reply:
One of these is returning the errors from the pickled file to the user? maybe i can remove the print statement in that function

Reply:
done